### PR TITLE
fix: RRSet save comments

### PR DESCRIPTION
**Pull request by bennetgallein**
## Description

Hello, while testing i realized that the code-side of updating/adding comments is there, just missing in the transformers for the actual request. This PR adds the functionality and should work as expected.

Relevant docs: https://doc.powerdns.com/authoritative/http-api/zone.html#rrset

## Motivation and context

Well

## How has this been tested?

./run-test script runs through no problem, changing my local development env to my github repo and adding/updating comments updates them accordingly

Include details of your testing environment, and the tests you ran to
see how your change affects other areas of the code, etc.

## Checklist:

Go over all the following points, and put an `x` in all the boxes that apply.

Please, please, please, don't send your pull request until all of the boxes are ticked. Once your pull request is created, it will trigger a build on our [continuous integration](http://www.phptherightway.com/#continuous-integration) server to make sure your [tests pass](https://help.github.com/articles/about-required-status-checks/).

- [x] I have read the **[CONTRIBUTING](../.github/CONTRIBUTING.md)** document.
- [x] My pull request addresses exactly one patch/feature.
- [x] My pull request contains a title that can be used as a release note.
- [x] I have created a branch for this patch/feature.
- [x] Each individual commit in the pull request is meaningful.
- [x] I have added tests to cover my changes.
- [x] If my change requires a change to the documentation, I have updated it accordingly.

If you're unsure about any of these, don't hesitate to ask. We're here to help!

### CODE REVIEW

File: tests/HelperTest.php [hunk 1]
```diff
@@ -13,13 +13,14 @@ class HelperTest extends TestCase
...
     {
-        $result = Helper::createResourceRecord('unit.test.', 'www', RecordType::A, '127.0.0.1', 1337);
+        $result = Helper::createResourceRecord('unit.test.', 'www', RecordType::A, '127.0.0.1', 1337, [['content' => 'Hello World', 'account' => 'Tester']]);
 
         self::assertSame('www.unit.test.', $result->getName());
         self::assertSame('A', $result->getType());
         self::assertSame(1337, $result->getTtl());
         self::assertCount(1, $result->getRecords());
         self::assertSame('127.0.0.1', $result->getRecords()[0]->getContent());
+        self::assertSame('Hello World', $result->getComments()[0]->getContent());
```
Comment (trizz) [inline tests/HelperTest.php:23]:
```suggestion
        self::assertSame('Hello World', $result->getComments()[0]->getContent());
        self::assertSame('Tester', $result->getComments()[0]->getAccount());
```

File: tests/HelperTest.php [hunk 2]
```diff
@@ -40,6 +53,7 @@ public function testWithArray(): void
         self::assertCount(2, $result->getRecords());
         self::assertSame('127.0.0.1', $result->getRecords()[0]->getContent());
         self::assertSame('127.0.0.2', $result->getRecords()[1]->getContent());
+        self::assertSame(111, $result->getComments()[1]->getModifiedAt());
```
Comment (trizz) [inline tests/HelperTest.php:56]:
```suggestion
        self::assertSame(111, $result->getComments()[1]->getModifiedAt());
        self::assertSame('World', $result->getComments()[1]->getContent());
        self::assertSame('rooti', $result->getComments()[1]->getAccount());
```

**Comment by trizz** [comment]
Hi @bennetgallein 

Thanks for your PR! It's looking good, but I've made some small suggestions in your tests because you forgot to add some assertions. Also, please run [php-cs-fixer](https://github.com/PHP-CS-Fixer/PHP-CS-Fixer) to [fix this files](https://github.com/exonet/powerdns-php/actions/runs/6552525914/job/17808895837?pr=144#step:4:6). Because this is a PR the required GitHub action can't commit to your fork.

**Comment by bennetgallein** [comment]
Relevant files are updated :+1: My auto-formatter wants to format differently so no saving + formatting for me :)

**Comment by bennet-esyoil** [comment]
ping ping @trizz :)

File: src/Zone.php [hunk 3]
```diff
@@ -21,25 +21,26 @@ class Zone extends AbstractZone
...
-     * @param string         $type    The type of the resource record.
-     * @param mixed[]|string $content The content of the resource record. When passing a multidimensional array,
-     *                                multiple records are created for this resource record.
-     * @param int            $ttl     The TTL.
+     * @param mixed[]|string $name     The resource record name.
+     * @param string         $type     The type of the resource record.
+     * @param mixed[]|string $content  The content of the resource record. When passing a multidimensional array,
+     *                                 multiple records are created for this resource record.
+     * @param array|mixed[]  $comments The comment to assign to the record.
+     * @param int            $ttl      The TTL.
```
Comment (robbinjanssen) [inline src/Zone.php:29]:
```suggestion
     * @param int            $ttl      The TTL.
     * @param array|mixed[]  $comments The comment to assign to the record.
```

**Review by robbinjanssen — CHANGES_REQUESTED** [review]

**Comment by robbinjanssen** [comment]
@bennetgallein i've created a discussion for your suggestion about the syntax, can you maybe provide a bit of pseudo code there of what you want to achieve? :)

File: src/Zone.php (hunk 3 above)
Comment (bennetgallein) [reply src/Zone.php]: Pushed!~

**Comment by robbinjanssen** [comment]
Published as v4.5.0! 👍🏻 Thanks!

**Pull request merged**